docs/blueprints/EP036.rst

```diff
@@ -0,0 +1,300 @@
+  - The chosen table group for each NApp should be defined (check Table Group section).
+
+Initially proposed here is a datastructured representing a linked list. 
+
+  .. code:: javascript
+
+    "pipeline": {
+      {
+        "id": "9cba7e27a33c45",
+        "status": "enabled",
+        "description": "First pipeline",
+        "multi_table": 
+        {
+          0: {
+            "table_miss_flow": {
+              "priority": 123, "match": {},
+              "instructions": [{"instruction_type": "goto_table", "table_id": 1}]
+            },
+            "napps_table_groups": {"coloring": ["base"],"of_lldp": ["base"]}
+          },
+          1: {
+            "table_miss_flow": {
+              "priority": 123, "match": {},
+              "instructions": [{"instruction_type": "goto_table","table_id": 2}]
+            },
+            "napps_table_groups": {"flow_manager": []}
+          },
+          2: {
+            "table_miss_flow": {
+              "priority": 123, "match": {},
+              "instructions": [{"instruction_type": "goto_table","table_id": 3}]
+            },
+            "napps_table_groups": {"mef_eline": ["evpl"]}
+          },
+          3: {
+            "table_miss_flow": {
+              "priority": 123, "match": {},
+              "instructions": [{"instruction_type": "goto_table","table_id": 5}]
+            },
+            "napps_table_groups": {"mef_eline": ["epl"]}
+          },
+          4: {
+            "table_miss_flow": {},
+            "napps_table_groups": {"telemetry_int": ["base"]}
+          },
+          5: {
+            "table_miss_flow": {},
+            "napps_table_groups": {}
+          }
+        }
+      }
+    }
+
+Events
+======
+
+Subscribed
+----------
+
+  - **kytos/flow_manager.flow.added**
+  - **kytos/of_core.handshake.completed**
+  - **kytos/flow_manager.flow.error**
+  - **kytos/[mef_eline|telemetry_int|coloring|of_lldp].enable_table** - Every NApp sends an event with their table group settings.
+
+kytos/<napp_name>.enable_table
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is sent from other NApps with information about their table settings.
+
+.. code:: javascript
+
+  "content": {
+    "group_table": {"evpl": 2, "epl": 3}
+  }
+
+Published
+---------
+
+kytos/of_multi_table.enable_table
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event should sent before any NApp sends its first flow to be published.
+
+.. code:: javascript
+
+  "content": {
+    "mef_eline": {"epl": 3, "evpl": 2},
+    "of_lldp": {"base": 0},
+    "coloring": {"base": 0}
+  }
+
+Endpoint API
+============
+  
+  - GET `/api/kytos/of_multi_table/v1/pipeline` - Get all pipelines in the database. Parameters enabled and disabled are allowed.
+
+  - GET `/api/kytos/of_multi_table/v1/pipeline/<pipeline_id>` - Get a single pipeline that matches the given pipeline id.
+
+  - POST `/api/kytos/of_multi_table/v1/pipeline/<pipeline_id>/enable` - Enable pipeline to be applied to all switches

```

Comment:
I've just realized a potential corner case here. Under normal circumstances, all pipelines creation with NApps and table groups will be validated, but there's not guarantee that each NApp are indeed enabled (and installed), and if `of_multi_table` will wait for the event then it'll never be responded. So either we have an event like `kytos/<napp_name>.enable_table`, being published as NApps are loaded too, otherwise you might need to make a request to `/api/kytos/core/napps_enabled` to check the NApps that are enabled. 

How are you envisioning this part?

Reply:
I will just check the controller.

Reply:
Sounds good. Let's close this thread then.